nexus/db-queries/src/db/queries/region_allocation.rs

```diff
@@ -86,47 +106,67 @@ struct CandidateDatasets {
 }
 
 impl CandidateDatasets {
-    fn new() -> Self {
+    fn new(allocation_strategy: &RegionAllocationStrategy) -> Self {
```

Comment:
Up to you -- but sometimes I pass the previous subquery as an explicit argument, to make the dependencies a little more obvious. It's obviously not necessary, but IMO it makes it a little more clear "what subqueries depend on what other subqueries".

For example, this subquery depends on the output of the `candidate_zpools` subquery.

Reply:
makes sense to me, I'll do it.